test/e2e/network/dns.go

```diff
@@ -313,7 +313,13 @@ var _ = SIGDescribe("DNS", func() {
 		validateTargetedProbeOutput(f, pod3, []string{wheezyFileName, jessieFileName}, svc.Spec.ClusterIP)
 	})
 
-	It("should support configurable pod resolv.conf", func() {
+	/*
+		Release: v1.14
+		Testname: DNS, dnsConfig
+		Description: Create a Pod as DNS_Server to serve DNS resolution by dnsName and IP. Create another pod B, with dnsConfig pointing to DNS_Server pod and DNS options. Provided dnsConfig entries MUST match with Pod B's /etc/resolve.conf file.
+		Pod MUST be able to resolve dnsName to its IP.
+	*/
+	framework.ConformanceIt("should support configurable pod resolv.conf", func() {
```

Comment:
This test does something pretty linux specific --- it checks if the `/etc/resolv.conf` file is generated in the expected way, which I think might not work on Windows as that is done via CNI DNS capabilities. Should we fix this behavior before promoting to conformance?

Reply:
This test should just be marked as LinuxOnly.

Reply:
Thanks for bringing clarity. I'm creating another PR to add [LinuxOnly] tag and we can hold this PR.

Reply:
Does windows support the custom dnsConfig via CNI? We should have a [sig-windows] test for that if so. It should not be a conformance test at this time, though, since Windows is an optional feature so is not strictly subject to conformance.

Reply:
Adding @feiskyer to this. Yeah IIUC Windows supports dnsConfig via CNI. One idea is we can also tweak this test to remove the linux specific step so that it can be re-used.

Reply:
@mgdevstack if it is possible to fix the test so that it works on both Linux and Windows, that is preferable to splitting it into two tests.

Reply:
I have not dug into CNI to implement DNS resolution with dnsConfig. Could we assign this to an appropriate member for the code update?
I agree to split this into two e2e due to implemention differences.

@johnbelamaric, As most of networking based e2e have different implementation for Windows/Linux, I would like to get clarity on
1. Should we have single e2e but call different implementations to verify specific behavior
e.g. E2E:  `should resolve DNS via dnsConfig`
```
if os != *nix
   windowsImplemention()
else
   nixImplemention()
Expect(err).NotTo(HaveOccured())
```
2. Should we have different e2e (might be having the same name but different tags/profile)
```
in test/e2e/network/dns.go
should resolve DNS via dnsConfig [LinuxOnly]
should resolve DNS via dnsConfig [WindowsOnly]
```
3. Which e2e are getting transitioned to `test/e2e/windows`
```
test/e2e/network
should resolve DNS via dnsConfig

test/e2e/windows
should resolve DNS via dnsConfig
```

Note: We may encounter duplicate entries (https://github.com/kubernetes/kubernetes/pull/75134#discussion_r266748023) of this e2e in Conformance.txt during Conformance promotion. (As tags are filtered out here)

Reply:
Indeed, AFAIK, this behaviour is implemented in Windows CNI, but there's no ``/etc/resolv.conf`` file. Instead, those configurations are set into registries, and can be retried through all sorts of commands, including ``ipconfig``.

So, at the very least, this should be ``[LinuxOnly]`` until the test also works on Windows.

Soo, I've been thinking. What if we build a binary and stick it into an image, so we shouldn't have to worry about them in the test cases? So, instead of calling or doing a thing that only works on Linux, it could also work on Windows?

For example, in this case, getting the DNS suffix list would be:

    /some_new_awesome_binary get dns-suffix-list

Or something.

Reply:
Hmm. Maybe. But that means a special image just for this test, which makes reviewing it hard, and that pattern could result in a proliferation of images. We should avoid it if possible.

Reply:
I didn't mean one image per test case, but one image for all test cases that would have different necessary commands for Linux and Windows. "One Image to rule them all," kind of Image.

Reply:
Yeah, that could work. I like the idea. There are a number of behaviors that probably would be easiest to validate - or can only be validated - by running a pod. For example, pod-to-pod networking when not in host networking mode. If we make sure that it works on all platforms/architectures (as described [here](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-requirements)), then we should be good to go with this.